firedrake/bcs.py

```diff
@@ -15,21 +15,14 @@
 import firedrake.utils as utils
 
 
-__all__ = ['DirichletBC', 'homogenize']
+__all__ = ['DirichletBC', 'homogenize', 'FormBC']
 
 
-class DirichletBC(object):
-    r'''Implementation of a strong Dirichlet boundary condition.
+class BCObject(object):
```

Comment:
Prefer `BCBase` or `AbstractBC` (I think in the rest of firedrake we typically use `FooBase` for these abstract base classes).